File: programmers/js_src/level00/120839.js
// 가위 바위 보
function solution(rsp) {
  var answer = '';

  rsp.split('').forEach(value => {
    if (value === '2') {
      answer += '0';
    } else if (value === '0') {
      answer += '5';
    } else {
      answer += '2';
    }
  });

  return answer;
}

// 다른 풀이
// function solution(rsp) {
//   let arr = {
//     2: 0,
//     0: 5,
//     5: 2
//   };

//   var answer = [...rsp].map(v => arr[v]).join("");

//   return answer;
// }

File: programmers/java/level00/120839.java
// 가위 바위 보

import java.util.HashMap;
import java.util.Arrays;
import java.util.stream.Collectors;

class Solution {
    public String solution(String rsp) {
        HashMap<String, String> map = new HashMap<>(){{
            put("2", "0");
            put("0", "5");
            put("5", "2");
        }};
        
        return Arrays.stream(rsp.split(""))
            .map(val -> map.get(val))
            .collect(Collectors.joining());
    }
}